tests/core/test-core.c

```diff
@@ -641,6 +726,27 @@ static void test_brick_core_verify_multiple_link(void)
 	east_brick = pg_brick_new("nop", config, &error);
 	g_assert(!error);
 
+	west_brick1 = pg_brick_new("nop", config1, &error);
+	g_assert(!error);
+	middle_brick1 = pg_brick_new("nop", config1, &error);
+	g_assert(!error);
+	east_brick1 = pg_brick_new("nop", config1, &error);
+	g_assert(!error);
+
+	west_brick2 = pg_brick_new("nop", config2, &error);
+	g_assert(!error);
+	middle_brick2 = pg_brick_new("nop", config2, &error);
+	g_assert(!error);
+	east_brick2 = pg_brick_new("nop", config2, &error);
+	g_assert(!error);
+
+	west_brick3 = pg_brick_new("nop", config3, &error);
+	g_assert(!error);
+	middle_brick3 = pg_brick_new("nop", config3, &error);
+	g_assert(!error);
+	east_brick3 = pg_brick_new("nop", config3, &error);
+	g_assert(!error);
```

Comment:
I don't think you destroy those brick, you you either need to call `pg_brick_ptrptr_destroy` at the end of the function or creating the variable using `pg_cleanup(pg_brick_ptrptr_destroy)`
example: 
```
pg_cleanup(pg_brick_ptrptr_destroy) struct pg_brick_config *config =
pg_brick_config_new("mybrick", 2, 2, PG_MULTIPOLE);
```
also 
```
pg_cleanup(pg_brick_ptrptr_destroy)
```
is kind of ugly, so maybe we can create a macro 
```
#define pg_autobrick pg_cleanup(pg_brick_ptrptr_destroy)
```